x-pack/heartbeat/monitors/browser/synthexec/synthexec_test.go

```diff
@@ -149,11 +149,25 @@ func TestRunBadExitCodeCmd(t *testing.T) {
 	})
 }
 
-func runAndCollect(t *testing.T, cmd *exec.Cmd, stdinStr string) []*SynthEvent {
+func TestRunTimeoutExitCodeCmd(t *testing.T) {
+	cmd := exec.Command("go", "run", "./main.go")
```

Comment:
I like this test, are we sure that there isn't a race here? I wonder if on fast systems that might execute fast enough to be flaky. It might be safer to add a sleep into that go program.

Reply:
Added a small timeout to the executable just in case